packages/react-vapor/src/utils/JSXUtils.tsx

```diff
@@ -0,0 +1,10 @@
+import * as React from 'react';
+import {renderToStaticMarkup} from 'react-dom/server';
+import * as s from 'underscore.string';
+
+export type JSXRenderable = JSX.Element | JSX.Element[] | string | number;
+
+export const getReactNodeTextContent = (node: React.ReactNode): string => {
+    const HTMLString = renderToStaticMarkup(<div>{node}</div>);
+    return s.stripTags(HTMLString).replace(/\s{2,}/g, ' ');

```

Comment:
The replace here simply converts whitespaces chunks of 2 occurences or more into a single whitespace.
```
'Pizza is  my favorite   meal.' --> 'Pizza is my favorite meal.'
```

Reply:
I'd replace with `s(HTMLString).stripTags().clean().value();` I think

Reply:
The typings don't support it, dunno why. It works at runtime though.
![image](https://user-images.githubusercontent.com/35579930/59684779-b775e180-91a8-11e9-826f-6ed74103af0c.png)